packages/main/src/Dialog.ts

```diff
@@ -340,7 +342,14 @@ class Dialog extends Popup {
 	}
 
 	get _role() {
-		return (this.state === ValueState.Error || this.state === ValueState.Warning) ? "alertdialog" : "dialog";
+		if (this.accessibleRole === PopupAccessibleRole.None) {
+			this.shadowRoot!.querySelector(".ui5-popup-root")?.removeAttribute("role");
+			this.shadowRoot!.querySelector(".ui5-popup-root")?.removeAttribute("aria-modal");
+		}
```

Comment:
This getter does too much. Imagine accessing `const r = dialog._role;`and seeing the state being suddenly mutated.